src/Sylius/Bundle/OrderBundle/Controller/OrderController.php

```diff
@@ -220,4 +236,22 @@ private function getParameterFromRequest(Request $request, string $key)
 
         return null;
     }
+
+    private function isInStock(PersistentCollection $orderItem, FormInterface $form): bool
+    {
+        $orderItemsCollection = $orderItem->toArray();
+
+        foreach ($orderItemsCollection as $key => $orderItem) {
+            $onHand = $orderItem->getVariant()->getOnHand();
+
+            $quantity = $form->get('items')->get((string) $key)->get('quantity')->getData();
+
+
+            if ($quantity > $onHand){
+                return true;
+            }
+        }
+
+        return false;
```

Comment:
Not all products are tracked, that is probably the reason you've added unneeded steps in other scenarios.